tests/utils/test_uri.py

```diff
@@ -37,6 +37,27 @@ def test_get_db_info_from_uri(server_uri, result):
     assert get_db_info_from_uri(server_uri) == result
 
 
+@pytest.mark.parametrize("hostname, experiment_id, run_id, workspace_id, result", [
+    ('https://www.databricks.com/', '19201', '2231', '12211',
+     'https://www.databricks.com/?o=12211#mlflow/experiments/19201/runs/2231'),
+    ('https://www.databricks.com/', '19201', '2231', None,
+     'https://www.databricks.com/#mlflow/experiments/19201/runs/2231'),
+    ('https://www.databricks.com/', '19201', '2231', '0',
+     'https://www.databricks.com/#mlflow/experiments/19201/runs/2231'),
+    ('https://www.databricks.com/', '19201', '2231', '0',
+     'https://www.databricks.com/#mlflow/experiments/19201/runs/2231'),
+    (None, '19201', '2231', '0', 'Exception'),
+    ('https://www.databricks.com/', None, '2231', '0', 'Exception'),
+    ('https://www.databricks.com/', '19201', None, '0', 'Exception'),
+])
+def test_construct_run_url(hostname, experiment_id, run_id, workspace_id, result):
+    if result == 'Exception':
+        with pytest.raises(MlflowException):
+            construct_run_url(hostname, experiment_id, run_id, workspace_id)
```

Comment:
probably cleaner to just separate it out into another test case like `test_construct_run_url_errors`

Reply:
yeah looking back on it, definitely agree. will update!